lib/sensu/server/process.rb

```diff
@@ -55,20 +90,29 @@ def update_client_registry(client, &callback)
       end
 
       # Set up the client keepalive consumer, keeping the Sensu client
-      # registry updated. The consumer receives JSON serialized client
-      # keepalives from the transport, parses them, and calls
+      # registry updated for validated clients. The consumer receives
+      # JSON serialized client keepalives from the transport, parses them,
+      # validates the client sending the messages, and then calls
       # `update_client_registry()` with the client data to update the
-      # registry. Transport message acknowledgements are used to
-      # ensure the client registry is updated successfully. Keepalive
-      # JSON parsing errors are logged.
+      # registry. Messages that do not pass client validation are ignored.
+      # Transport message acknowledgements are used to ensure the client
+      # registry is updated successfully. Keepalive JSON parsing errors are
+      # logged.
       def setup_keepalives
         @logger.debug("subscribing to keepalives")
         @transport.subscribe(:direct, "keepalives", "keepalives", :ack => true) do |message_info, message|
           @logger.debug("received keepalive", :message => message)
           begin
             client = MultiJson.load(message)
-            update_client_registry(client) do
```

Comment:
Should move `validate_client()` within `update_client_registry()` to keep `setup_keepalives()` simple and easy to follow.